### ISSUE TITLE
Add ability to disable documentation of fields

### ISSUE THREAD

**Issue by ryobkin** [DotNetAnalyzers/StyleCopAnalyzers#3116]
Older versions of StyleCop appear to have had an option to disable the requirement for documentation of fields.
Settings.StyleCop -> Documentation Rules -> Include fields

Would it be possible to include this feature in StyleCop Analyzers?

(Referenced here and in other places)
https://stackoverflow.com/questions/1095390/the-field-must-have-a-documentation-header-style-cop-code-smell

**Comment by sharwell** [comment]
This is already the default configuration:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md#documentation-requirements

**Comment by ryobkin** [comment]
The option documentPrivateFields only controls whether documentation is required for private fields. The old "Include fields" option was enabled by default but when disabled it would remove the requirement to document any fields regardless of access modifiers.

**Comment by sharwell** [comment]
My recommendation for that scenario would be the following three things:

1. Use the baselining feature to suppress all current diagnostics in GlobalSuppressions.cs
2. Avoid using non-private fields
3. Document new non-private fields if/when added

This approach should minimize ongoing maintenance costs without relying on further complications to the rules themselves. Would that work for you?

**Comment by ryobkin** [comment]
Not really. We have a large number of public or internal fields that are constant or static readonly. While baselining would deal with the existing ones we are sure to add more.

Most of my team feels that fields should only have documentation if they do something unexpected. Most of the fields in question are either const or static readonly and are already well named. There is also a concern that adding a requirement to document new fields may lead to less maintainable code - such as multiple private copies of the same constant in different classes.
